spec/services/teachers/search_spec.rb

```diff
@@ -147,6 +114,64 @@
       end
     end
 
+    describe "searching for ECTs at a school" do
+      context 'when one school is present' do
+        it 'only selects ECTs who are currently at the given school' do
+          query = Teachers::Search.new(ect_at_school: 123).search
+
+          expect(query.to_sql).to include(%("ect_at_school_periods"."school_id" = 123))
+        end
+
+        it 'only selects ongoing ECT at school periods' do
+          query = Teachers::Search.new(ect_at_school: 123).search
+
+          expect(query.to_sql).to include(%("ect_at_school_periods"."finished_on" IS NULL))
+        end
+      end
+
+      context 'when multiple schools are present' do
+        it 'only selects ECTs who are currently at the given school' do
+          query = Teachers::Search.new(ect_at_school: [123, 456]).search
+
+          expect(query.to_sql).to include(%{"ect_at_school_periods"."school_id" IN (123, 456)})
+        end
+      end
+
+      context 'when absent' do
+        it 'does not join ect_at_school_periods' do
+          query = Teachers::Search.new(ect_at_school: 123).search
+
+          expect(query).not_to include('ect_at_school_periods')
```

Comment:
I think you might be getting false positives here because your missing the `to_sql`, should it be:


```suggestion
        it 'does not join ect_at_school_periods' do
          query = Teachers::Search.new.search

          expect(query.to_sql).not_to include('ect_at_school_periods')
```

Reply:
Ah well caught, thanks!